tests/ssg_test_suite/test_env.py

```diff
@@ -248,40 +227,152 @@ def reset_state_to(self, state_name, new_running_state_name):
+                .format(str(exc)))
+            raise RuntimeError(msg)
+
+    def _commit(self, container, image):
+        container.commit(repository=image)
 
     def _new_container_from_image(self, image_name, container_name):
-        img = self._find_image_by_name(image_name)
+        img = self.client.images.get(image_name)
         result = self.client.containers.run(
             img, "/usr/sbin/sshd -D",
             name="{0}_{1}".format(self._name_stem, container_name), ports={"22": None},
             detach=True)
         return result
 
+    def _get_container_ip(self, container):
+        container.reload()
+        container_ip = container.attrs["NetworkSettings"]["Networks"]["bridge"]["IPAddress"]
+        return container_ip
+
     def _terminate_current_running_container_if_applicable(self):
         if self.containers:
             running_state = self.containers.pop()
             running_state.stop()
             running_state.remove()
 
-    def _delete_saved_state(self, image):
-        self._terminate_current_running_container_if_applicable()
-
-        assert self.created_images
-
-        associated_image = self.created_images.pop()
-        assert associated_image == image
-        self.client.images.remove(associated_image)
-
-    def discard_running_state(self, state_handle):
-        self._terminate_current_running_container_if_applicable()
+    def _remove_image(self, image):
+        self.client.images.remove(image)
 
     def _local_oscap_check_base_arguments(self):
         return ['oscap-docker', "container", self.current_container.id,
                                                             'xccdf', 'eval']
 
-    def offline_scan(self, args, verbose_path):
-        command_list = self._local_oscap_check_base_arguments() + args
 
-        return common.run_cmd_local(command_list, verbose_path)
+class PodmanTestEnv(ContainerTestEnv):
+    # TODO: Rework this class using Podman Python bindings (python3-podman)
+    # at the moment when their API will provide methods to run containers,
+    # commit images and inspect containers
+    name = "podman-based"
+
+    def __init__(self, scanning_mode, image_name):
+        super(PodmanTestEnv, self).__init__(scanning_mode, image_name)
+
+    def _commit(self, container, image):
+        podman_cmd = ["podman", "commit", container, image]
+        try:
+            subprocess.check_output(podman_cmd, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            msg = "Command '{0}' returned {1}:\n{2}".format(" ".join(e.cmd), e.returncode, e.output.decode("utf-8"))
+            raise RuntimeError(msg)
+
+    def _new_container_from_image(self, image_name, container_name):
+        long_name = "{0}_{1}".format(self._name_stem, container_name)
+        podman_cmd = ["podman", "run", "--name", long_name,
+                      "--publish", "22", "--detach", image_name,
+                      "/usr/sbin/sshd", "-D"]
```

Comment:
This is rather a question, but do we really need to override the `CMD` from the `Dockerfile`? We don't do that in the Docker case.

Reply:
We do that in Docker case:
https://github.com/ComplianceAsCode/content/blob/c3178080fc69045a1ea5dd4cd35be79d0bf33023/tests/ssg_test_suite/test_env.py#L256-L259

Moreover, our Dockerfiles do not specify `CMD`, therefore if we use our Dockerfiles, we aren't overriding anything.

Reply:
Damn, you are right!